docs/core/extensions/logging-library-authors.md

```diff
@@ -0,0 +1,85 @@
+---
+title: Logging guidance for .NET library authors
+author: IEvangelist
+description: Learn how to expose logging as a library author in .NET. Follow the guidance to ensure your library is correctly exposed to consumers.
+ms.author: dapine
+ms.date: 08/03/2023
+---
+
+# Logging guidance for .NET library authors
+
+As a library author, exposing logging is a great way to provide consumers with insight into the inner workings of your library. This guidance helps you expose logging in a way that is consistent with other .NET libraries and frameworks.
+
+## Use the `ILoggerFactory` interface
+
+When your library exposes configuration functionality for logging, it might be tempting to accept an <xref:Microsoft.Extensions.Logging.ILogger>, but that isn't recommended as it represents a single category. An <xref:Microsoft.Extensions.Logging.ILogger%601> represents a one-to-one relationship between a category and a type. If your library allows consumers to provide one of these interfaces, it means that you're limiting the consumer to a single category, which may not be appropriate for their use case.
```

Comment:
As-is this guidance may seem confusing if the library author uses DI because the guidance here is saying use ILoggerFactory and guidance elsewhere is saying use ILogger\<T> in the class constructor.

Perhaps something like this? (Feel free not to use this wording or weave in examples, I'm just trying to convey the general idea)

## When to use ILoggerFactory vs. ILogger\<T>

When writing a library that emits logs, you will need an ILogger object to record the logs. To get that object your API can either accept an ILogger\<T> parameter, or it can accept an ILoggerFactory after which you call ILoggerFactory.CreateLogger(). Which approach should be prefered?
- If you need a logging object that can be passed along to multiple classes so that all of them can emit logs, use ILoggerFactory. It is recommended that each class create logs with a separate category, named the same as the class name. In order to do this you will need the factory to create unique ILogger\<T> objects for each class that is emitting logs. Common examples of this would be public entrypoint APIs for a library or public constructors of types that internally may create helper classes.
- If you need a logging object that will only be used inside one class and never shared use ILogger\<T> where T is the type that will be producing the logs. A common example of this is a constructor for a class created by dependency injection.

If you are designing a public API that must remain stable over time, keep in mind that you might desire to refactor your internal implementation in the future. Even if a class doesn't create any internal helper types initially, that might change as the code evolves. Using ILoggerFactory will accomodate creating new ILogger\<T> objects for any new classes without changing the public API.